### ISSUE TITLE
Reduce duplicate logic in admin-notices.php

### ISSUE THREAD

**Issue by cklosowski** [easydigitaldownloads/Easy-Digital-Downloads#2707]
Looking at the ```includes/admin/admin-notices.php``` file there is a TON of code duplication that could be simplified into a single call. I'm thinking a helper function like:

```edd_show_admin_message()``` and you can provide 2 parameters:

```
$message_id - a string like 'discount_added' or 'payment_deleted'
$capability - The cap to check when verifying we should display this;
```

It could look something like:
```
function edd_show_admin_message( $message_id = false, $capability = false) {

    if ( ! isset( $_GET['edd-message'] ) {
        return false;
    }

    if ( $message_id !== $_GET['edd-message'] ) {
        return false;
    }

    if ( ! current_user_can( $capability ) ) {
        return false;
    }

    return true;

}
```

**Comment by pippinsplugins** [comment]
This looks great to me. Everything works great.

**Comment by pippinsplugins** [comment]
Looks like this was never merged into release/2.3

**Comment by pippinsplugins** [comment]
Nevermind, yes it was, the old file just wasn't deleted.
